lib/hackerSortingAlgo.ts

```diff
@@ -0,0 +1,103 @@
+const fs = require('fs')
+const path = require('path')
+const { parse } = require('csv-parse/sync')
+
+// parse csv file into an array
+function loadCSV (filepath: string): any[] {
+  const csvFileAbsolutePath = path.resolve(__dirname, 'data', filepath)
+
+  // error handling in case file is missing
+  let fileContent
+  try {
+    fileContent = fs.readFileSync(csvFileAbsolutePath, {
+      encoding: 'utf-8'
+    })
+  } catch (err) {
+    console.log(`File cannot be found : "${filepath}"`)
+    return []
+  }
+
+  const options = {
+    delimiter: ',',
+    columns: true
+  }
+  return parse(fileContent, options)
+}
+
+// loops through each user row in the given array
+// -> for each question: increment count for corresponding cabin if answers match
+function matchAnswers (hackerList: any[], answerList: any[], cabinList: any[]) {
+  // constants for cabin size and number of questions
+  const CABIN_SIZE = Object.keys(cabinList).length
+  const QUESTIONS_SIZE = Object.keys(answerList[0]).length
+  hackerList.forEach((hacker: any, hackerIndex: number) => {
+    // each element = a different cabin, all initialized to 0
+    const counter = Array<number>(CABIN_SIZE).fill(0)
```

Comment:
In order to give more meaning to variable names, maybe we could call `counter` `cabinScore` or something similar!